controllers/usersignup/usersignup_controller.go

```diff
@@ -200,12 +217,191 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
+			}
+
+			// Create the migrated UserSignup here.
+			//
+			// We need to:
+			//
+			// 1) Copy the existing UserSignup
+			// 2) Override the new UserSignup resource name with the encoded username
+			// 3) Set the starting status to Deactivated (technically we could let the reconciler function do this
+			//    when it reconciles the migrated UserSignup, but it shouldn't hurt to set this status up front)
+			migratedUserSignup = userSignup.DeepCopy()
+			migratedUserSignup.ObjectMeta.ResourceVersion = ""
+			migratedUserSignup.ObjectMeta.Generation = 0
+			migratedUserSignup.ObjectMeta.CreationTimestamp = v1.Time{}
+			migratedUserSignup.Name = encodedUsername
+			migratedUserSignup.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(migratedUserSignup.Status.Conditions,
+				toolchainv1alpha1.Condition{
+					Type:    toolchainv1alpha1.UserSignupComplete,
+					Status:  corev1.ConditionTrue,
+					Reason:  toolchainv1alpha1.UserSignupUserDeactivatedReason,
+					Message: "",
+				})
+			migratedUserSignup.Annotations[migrationAnnotationName] = userSignup.Name
+			err = r.Client.Create(context.TODO(), migratedUserSignup)
+			if err != nil {
+				// If there was an error creating the migrated UserSignup, then set the status and requeue
+				return r.wrapErrorWithStatusUpdate(logger, userSignup,
+					r.setStatusMigrationFailedCreate, err, "Failed to migrate UserSignup")
+			}
+		}
+	}
+	return nil
+}
+
+// cleanupMigration performs migration cleanup if the "migration-replaces" annotation has been set on the UserSignup
+//
+// If the annotation has been set, then it indicates that this UserSignup has been migrated, and that the
+// *original* UserSignup should be deleted.  The value of the annotation is the name of the original UserSignup
+// resource that should now be deleted.
+//
+// This function should:
+//
+// 1) Delete the original UserSignup, then
+// 2) Remove the migration-replaces annotation
+func (r *Reconciler) cleanupMigration(userSignup *toolchainv1alpha1.UserSignup, originalUserSignupName string,
+	request ctrl.Request, logger logr.Logger) (ctrl.Result, error) {
+
+	userSignupToDelete := &toolchainv1alpha1.UserSignup{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: request.Namespace,
+		Name:      originalUserSignupName,
+	}, userSignupToDelete)
+	if err == nil {
+		// Delete the original UserSignup, if it has finished deactivating
+		cond, found := condition.FindConditionByType(userSignupToDelete.Status.Conditions, toolchainv1alpha1.UserSignupComplete)
+		if !found || cond.Reason != toolchainv1alpha1.UserSignupUserDeactivatedReason {
+			// The UserSignup to delete isn't finished deactivating yet, return an error
+			return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup, r.setStatusMigrationFailedCleanup,
+				errs.New("Original UserSignup not deactivated"),
+				fmt.Sprintf("Original UserSignup [%s] not yet deactivated", userSignupToDelete.Name))
+		}
```

Comment:
wouldn't make more sense to check the migration condition - that it's true because it successfully created the new UserSignup resource?